# ==============================================================================
# REGRESSOR COMMIT DIFF
# ==============================================================================
# File: js/src/vm/MemoryMetrics.cpp
# Commit: e9b797f6d7db
# Full Hash: e9b797f6d7db42f5ee5c0023ce5fc020759a5e4c
# Author: Jon Coppeard <jcoppeard@mozilla.com>
# Date: 2020-10-13 21:43:51
# Regressor Bug: 1657025
# File Overlap Count: 1
# Description:
#   Bug 1657025 - Wait for all background GC tasks to finish before collecting memory stats r=sfink
#   
#   This adds a new function to wait for all background tasks after a GC has
#   finished. It does remove the wait on background free task from FinishGC but I
#   couldn't see anywhere that would matter and try is green with this patch.
# ==============================================================================

diff -r d338a3af3f80 -r e9b797f6d7db js/src/vm/MemoryMetrics.cpp
--- a/js/src/vm/MemoryMetrics.cpp	Tue Oct 13 10:16:46 2020 +0000
+++ b/js/src/vm/MemoryMetrics.cpp	Mon Oct 12 19:03:20 2020 +0000
@@ -625,6 +625,7 @@
   // Finish any ongoing incremental GC that may change the data we're gathering
   // and ensure that we don't do anything that could start another one.
   gc::FinishGC(cx);
+  gc::WaitForBackgroundTasks(cx);
   JS::AutoAssertNoGC nogc(cx);
 
   JSRuntime* rt = cx->runtime();
